### PR TITLE
[23.05] cups: 2.4.2 -> 2.4.5

### PR DESCRIPTION
**Pull request by onny**
###### Description of changes

See release notes for 2.4.3: https://github.com/OpenPrinting/cups/releases/tag/v2.4.3

- Includes criticial security fix
- Many usefull fixes which should be available on stable release
- 2.4.4-2.4.5 only includes minor hotfixes

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vcunat** [comment]
I don't see any security fix.  Only including the one that was already patched.

**Comment by onny** [comment]
> I don't see any security fix. Only including the one that was already patched.

oh sorry this is what i meant :O

**Comment by fabianhjr** [comment]
https://nvd.nist.gov/vuln/detail/CVE-2022-26691 (CVSS: 6.7)

**Comment by fabianhjr** [comment]
Oh nvm had a brainfart, CVE-2022-26691 is up-to excluding 2.4.2, 2.4.2 is unaffected.

File: pkgs/misc/cups/default.nix [hunk 1]
```diff
@@ -24,24 +24,15 @@
...
   src = fetchurl {
     url = "https://github.com/OpenPrinting/cups/releases/download/v${version}/cups-${version}-source.tar.gz";
-    sha256 = "sha256-8DzLQLCH0eMJQKQOAUHcu6Jj85l0wg658lIQZsnGyQg=";
+    sha256 = "sha256-mkBN5V90UlsKaFHfDP3r+hIVrsDnwve+a5sJtpFvsAA=";
   };
 
   outputs = [ "out" "lib" "dev" "man" ];
 
-  patches = [
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/misc/cups/default.nix:37]: It was forgotten in the original commit to drop fetchpatch from inputs

**Comment by adamcstephens** [comment]
v2.4.6 is out now.

**Pull request merged**